drivers/usb/udc/udc_ambiq.c

```diff
@@ -0,0 +1,915 @@
+#include <soc.h>
+#include "am_mcu_apollo.h"
+#include <string.h>
+#include <zephyr/drivers/clock_control/clock_control_ambiq.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/usb/udc.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/cache.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#include "udc_common.h"
+
+LOG_MODULE_REGISTER(udc_ambiq, CONFIG_UDC_DRIVER_LOG_LEVEL);
+
+/* UDC event ID enumeration */
+enum udc_ambiq_event_type {
+	/* SETUP packet received at Control Endpoint */
+	UDC_AMBIQ_EVT_HAL_SETUP,
+	/* OUT transaction completed */
+	UDC_AMBIQ_EVT_HAL_OUT_CMP,
+	/* IN transaction completed */
+	UDC_AMBIQ_EVT_HAL_IN_CMP,
+	/* Xfer request received via udc_ambiq_ep_enqueue API */
+	UDC_AMBIQ_EVT_XFER,
+};
+
+/* UDC event queue element structure */
+struct udc_ambiq_event {
+	const struct device *dev;
+	enum udc_ambiq_event_type type;
+	uint8_t ep;
+};
+
+/* Define message queue for UDC events */
+K_MSGQ_DEFINE(drv_msgq, sizeof(struct udc_ambiq_event), CONFIG_UDC_AMBIQ_MAX_QMESSAGES,
+	      sizeof(void *));
+
+/* USB device controller access from devicetree */
+#define DT_DRV_COMPAT ambiq_usb
+
+/* Max Packet Size for USB endpoints */
+#define EP0_MPS   64U
+#define EP_FS_MPS 64U
+#define EP_HS_MPS 512U
+
+/* UDC driver private data structure definition */
+struct udc_ambiq_data {
+	struct k_thread thread_data;
+	void *usbHandle;
+	am_hal_usb_dev_speed_e eUsbSpeed;
+	uint8_t setup[8];
+};
+
+/* UDC driver configuration structure definition */
+struct udc_ambiq_config {
+	uint32_t num_endpoints;
+	int speed_idx;
+	struct udc_ep_config *ep_cfg_in;
+	struct udc_ep_config *ep_cfg_out;
+	void (*make_thread)(const struct device *dev);
+	void (*irq_enable_func)(const struct device *dev);
+	void (*irq_disable_func)(const struct device *dev);
+};
+
+/* Static variables */
```

Comment:
Please cleanup your code and remove all the unnecessary comments like `/* Static variables */`.

Reply:
Hi @jfischer-no, it seems my comment style is too verbose, removed all the comments that is there just to describe what a struct/function/global-variable does. Please let me know if my style is still too verbose. Thanks.